### PR TITLE
Scrum 1244 api mod association

### PR DESCRIPTION
**Pull request by FB-ping**
new endpoint for mod_corpus_associations

### CODE REVIEW

File: backend/app/literature/schemas/reference_category_schemas.py [hunk 1]
```diff
@@ -14,3 +14,4 @@ class ReferenceCategory(str, Enum):
     Internal_Process_Reference = "internal_process_reference"
     Unknown = "unknown"
     Retraction = "retraction"
+    Test = "test"
```
Comment (azurebrd) [inline backend/app/literature/schemas/reference_category_schemas.py:17]: Hi @FB-ping  This probably shouldn't be committed.

**Review by azurebrd — COMMENTED** [review]
HI @FB-ping   
@valearna rolled back FB-server changes to the .env and the alembic.ini
There's an extra  Test entry in [backend/app/literature/schemas/reference_category_schemas.py](https://github.com/alliance-genome/agr_literature_service/pull/83/files#diff-85cfd2704efd6b6f1ef77e7c560f9f2fd44095ed2f789901b1955b3cd8e34a1c) that probably shouldn't be there.
What is the change for [alembic/versions/226a02bfa5b3_change_author_cross_reference_cascade.py](https://github.com/alliance-genome/agr_literature_service/pull/83/files#diff-01309397047dc778ad80d10cc34df96fc7d7e40df4b3bf6ef48e16d445880858) ? I thought you found that for mod_corpus_association because they are only new tables, there didn't need to be any alembic changes.
What is test_resources_delete.py  is that a new unittest or something that we need in the repo, or were you making tests on how _version tables work, and we may or may not want them in the repo ?
What does the change on [backend/app/literature/models/cross_reference_model.py](https://github.com/alliance-genome/agr_literature_service/pull/83/files#diff-4ed7b0438ceb00c0677b0cf74f6d38081ce78bc82a14f3164c1226253f2bfd8c) to ondelete CASCADE do ?  Does it delete cross_references when the reference is deleted (good) or does it delete references when a cross_reference is deleted (bad) ?
Same question for author_model.  I'm guessing it's the correct thing (delete author + cross_reference when the reference they're attached to gets deleted)
Thanks Pinglei !

**Comment by azurebrd** [comment]
@valearna has an API running this code https://dev4004-literature-rest.alliancegenome.org/docs pointing to rds dev, for anyone that wants to test things.

File: backend/app/literature/schemas/reference_category_schemas.py (hunk 1 above)
Comment (FB-ping) [reply backend/app/literature/schemas/reference_category_schemas.py:17]: @azurebrd  I didn't touch this file.

Comment (azurebrd) [reply backend/app/literature/schemas/reference_category_schemas.py:17]: Oh, that's weird, thanks Pinglei.  Before we merge, we should roll it back.

**Comment by azurebrd** [comment]
@FB-ping @valearna Did tests on the API at https://dev4004-literature-rest.alliancegenome.org/docs
and made a comment https://agr-jira.atlassian.net/browse/SCRUM-1244?focusedCommentId=34017  with something that needs to be fixed for this ticket, something to be fixed about the author endpoint, and most everything else looks good.

File: backend/app/literature/schemas/reference_category_schemas.py (hunk 1 above)
Comment (azurebrd) [reply backend/app/literature/schemas/reference_category_schemas.py]: [5913f96](https://github.com/alliance-genome/agr_literature_service/pull/83/commits/5913f96ec90687d56d90c63a0992a3d3d5dce1a2) fixes that

**Comment by valearna** [comment]
Apart from the functional issues that @azurebrd requested on the ticket, all the rest looks good to me.

**Comment by valearna** [comment]
We merged these changes from another branch. Can we close this PR @FB-ping @azurebrd @nuin ?

**Comment by azurebrd** [comment]
> We merged these changes from another branch. Can we close this PR @FB-ping @azurebrd @nuin ?

Sounds good to me.  I don't know how much @FB-ping is looking at messages, so I don't know that he's going to see this anytime soon.  I guess we could wait a couple of days to see if he does.  Or we could close it but not delete it ?

**Comment by FB-ping** [comment]
no long needed

**Pull request closed without merging**